bothan-lib/src/telemetry/metrics.rs

```diff
@@ -0,0 +1,61 @@
+use opentelemetry::{global, KeyValue};
+
+pub struct Metrics{}
+
```

Comment:
instead of calling internal functions from metric, it'd probably be better if metrics kept some of its own variables (like source, meter, specific counters/histogram). building a new counter/histogram on each call doesn't seem very efficient. Alternatively, we could just keep each instrument as a pub value in the struct and let the users call it themselves